packages/gatsby-theme-carbon/src/components/Code/Code.js

```diff
@@ -23,25 +23,27 @@ const Code = ({ children, className: classNameProp = '', path, src }) => {
   const language = classNameProp.replace(/language-/, '').replace('mdx', 'jsx');
 
   const removeTrailingEmptyLine = (lines) => {
-    const [lastLine] = lines.splice(-1);
-    if (lastLine[0].empty) {
-      return lines;
+    if (lines && lines.length) {
```

Comment:
These checks are almost random that I applied to make the build work as this component could not find lines in a few pages. It would make sense to look into this as a separate PR.